util/auto_roll_logger.cc

```diff
@@ -44,6 +84,69 @@ void AutoRollLogger::RollLogFile() {
     now++;
   } while (env_->FileExists(old_fname).ok());
   env_->RenameFile(log_fname_, old_fname);
+  old_log_files_.push(old_fname);
+}
+
+void AutoRollLogger::GetExistingFiles() {
+  // Empty the queue to avoid duplicated entires in the queue.
+  while (!old_log_files_.empty()) {
+    old_log_files_.pop();
+  }
+
+  std::vector<std::string> all_files;
+  std::string parent_dir = db_log_dir_.empty() ? dbname_ : db_log_dir_;
+  Status s = env_->GetChildren(parent_dir, &all_files);
+  if (status_.ok()) {
+    status_ = s;
+  }
+  InfoLogPrefix info_log_prefix(!db_log_dir_.empty(), dbname_);
+  std::vector<std::string> info_log_files;
+  for (const std::string& f : all_files) {
```

Comment:
can you use `GetInfoLogList` from https://github.com/facebook/rocksdb/blob/master/utilities/convenience/info_log_finder.cc#L16?

Reply:
We don't have `DB*` here, so we can't use it. But let me see whether I can create a shared function to consolidate the code.